### PR TITLE
chore: bump status-go

### PR DESCRIPTION
**Pull request by osmaczko**
To test: https://github.com/status-im/status-desktop/issues/14914

### CODE REVIEW

**Comment by osmaczko** [comment]
@status-im/desktop-qa could you please test members reevaluation flows with collectibles please?

**Comment by anastasiyaig** [comment]
@osmaczko i gave it a short spin today with @virginiabalducci and @lukaszso and want to clarify some things just now with you: 

1. i was a member of community that does not require request to join and there was a view only channel permission set. Then new permission was added, requires me to keep a collectible to become a member. I received a notification that permission was added, but I was not kicked from community right after the permission to become a member was created (no re-evaluation? Shall i be kicked in this case?) . When clicking Shared addresses, it says that i am not eligible to be a member. Screenshot below

Screenshot 2024-06-05 at 15 08 33

2. Token master is not treated as token master without application restart

3.  When trying to re-join as a member to a community that has view only channel + a permission to keep a token to become a member, I can't spectate the community. Is that expected?

Screenshot 2024-06-05 at 15 19 44

image

4. if i keep Shared addresses screen opened and i am kicked from the community while keep the modal opened, the modal is not destroyed. I want to clarify if we want to fix it (in master for sure) by UI team? 

Screenshot 2024-06-05 at 15 08 33

**Comment by osmaczko** [comment]
> 1. i was a member of community that does not require request to join and there was a view only channel permission set. Then new permission was added, requires me to keep a collectible to become a member. I received a notification that permission was added, but I was not kicked from community right after the permission to become a member was created (no re-evaluation? Shall i be kicked in this case?) . When clicking Shared addresses, it says that i am not eligible to be a member. Screenshot below

The re-evaluation may happen with a 5min delay and the behavior you are seeing is expected, you'll be kicked eventually.

> Token master is not treated as token master without application restart

Application restart of whom?  Control node or token master? If the former, could it be similar case as above - token master not being evaluated yet? If the latter, then it is token-master client side issue, most likely not related to my changes.

> When trying to re-join as a member to a community that has view only channel + a permission to keep a token to become a member, I can't spectate the community. Is that expected?

Since the community is on the section list, it means you are in spectate mode, at least. It is expected to not see content if you don't have permissions to become member.
Btw. I think request to join community should be disabled if you don't meet the requirements.. :thinking: (not related to my changes)

> if i keep Shared addresses screen opened and i am kicked from the community while keep the modal opened, the modal is not destroyed. I want to clarify if we want to fix it (in master for sure) by UI team?

I don't think this is a big issue. What could happen in the worst-case scenario? Also, that's a very rare edge case for a normal user. In my opinion, it's not worth the burden of extra modal complexity.

**Comment by osmaczko** [comment]
@anastasiyaig, is it good to go or you need a bit more testing/help ?

**Comment by anastasiyaig** [comment]
i would give it a bit more try tonight (also @virginiabalducci and @lukaszso are testing now). Can it wait for tomorrow?

**Comment by virginiabalducci** [comment]
Hi @osmaczko 

@lukaszso and me tested several user flows using different combination of permissions (community level permission, user level permission and channel level permissions) found two issues:

**Issue 1**
**Members eligible to see token-restricted channel do not have access to it.** 
Channel permissions: Anyone who holds specific token is allowed to View and Post. Channel is hidden from members who don’t have the permissions to view channel.
Member holding token is unable to see channel. When checking the addresses this member shared with the community, the channel is there. But member cannot access it. 
This is reproducible on all the accounts we tested that owned the specific token.


Member with token master screen:
Screenshot 2024-06-06 at 10 43 39 AM

Channels member should be able to see (screen taken from Owner's account)
Screenshot 2024-06-06 at 10 36 53 AM

Permission of token-restricted channel
Screenshot 2024-06-06 at 10 17 36 AM

**Issue 2**

**Holder of TMaster token is unable to access community admin permissions and is treated as a regular member.** I was able to reproduce this on my end but it's working as expected on Lukasz end. I wonder if this may be specific to some configuration on my test accounts? Airdropped a TokenMaster to **two** different test accounts and still do not see the account as an Admin.

Comparison between Member (who holds TokenMaster) and Owner, notice the difference in the channel list

https://github.com/status-im/status-desktop/assets/68916150/8c1ad6b3-9c7d-4a2f-8142-183a60fda3df

Sharing console log in case it may help. Log triggered when viewing the user's shared addresses with the community. 
```DBG 2024-06-06 12:21:48.217-03:00 NewBE_callPrivateRPC                       topics="rpc" tid=1592300 file=core.nim:27 rpc_method=wakuext_slowdownArchivesImport
DBG 2024-06-06 12:21:51.331-03:00 NewBE_callPrivateRPC                       topics="rpc" tid=1592300 file=core.nim:27 rpc_method=wakuext_speedupArchivesImport
DBG 2024-06-06 12:21:55.115-03:00 NewBE_callPrivateRPC                       topics="rpc" tid=1592300 file=core.nim:27 rpc_method=wakuext_slowdownArchivesImport
DBG 2024-06-06 12:21:55.213-03:00 NewBE_callPrivateRPC                       topics="rpc" tid=1592300 file=core.nim:27 rpc_method=wakuext_getContactByID
DBG 2024-06-06 12:21:55.214-03:00 id doesn't have expected length            topics="contacts-service" tid=1592300 file=service.nim:417
DBG 2024-06-06 12:21:55.214-03:00 [threadpool task thread] initiating task   topics="task-threadpool" tid=1592320 file=threadpool.nim:54 messageType=AsyncGetRevealedAccountsArg:ObjectType threadid=1592320 task="{\"$type\":\"AsyncGetRevealedAccountsArg:ObjectType\",\"communityId\":\"0x021fbe1eebae9552401e7afb307c8a9cf78d1131ab4a76ef4635838e907f034eda\",\"memberPubkey\":\"0x040ad366b04230dea52b886d891047a184c196f08f93ee74dfd203a2e4b9978721a2fcd8cf91dbad239f6dc0d2f2bcb957efae0b8bad92fe1244eecbc62d92643d\",\"vptr\":105553181865472,\"slot\":\"onAsyncGetRevealedAccountsForMemberCompleted\",\"tptr\":4309894576}"
DBG 2024-06-06 12:21:55.214-03:00 NewBE_callPrivateRPC                       topics="rpc" tid=1592320 file=core.nim:27 rpc_method=wakuext_getRevealedAccounts
DBG 2024-06-06 12:21:55.214-03:00 [threadpool task thread] initiating task   topics="task-threadpool" tid=1592317 file=threadpool.nim:54 messageType=AsyncCheckPermissionsToJoinTaskArg:ObjectType threadid=1592317 task="{\"$type\":\"AsyncCheckPermissionsToJoinTaskArg:ObjectType\",\"communityId\":\"0x021fbe1eebae9552401e7afb307c8a9cf78d1131ab4a76ef4635838e907f034eda\",\"addresses\":[],\"vptr\":105553181865472,\"slot\":\"onAsyncCheckPermissionsToJoinDone\",\"tptr\":4309784232}"
DBG 2024-06-06 12:21:55.214-03:00 [threadpool task thread] initiating task   topics="task-threadpool" tid=1592316 file=threadpool.nim:54 messageType=AsyncCheckAllChannelsPermissionsTaskArg:ObjectType threadid=1592316 task="{\"$type\":\"AsyncCheckAllChannelsPermissionsTaskArg:ObjectType\",\"communityId\":\"0x021fbe1eebae9552401e7afb307c8a9cf78d1131ab4a76ef4635838e907f034eda\",\"addresses\":[],\"vptr\":105553181865216,\"slot\":\"onAsyncCheckAllChannelsPermissionsDone\",\"tptr\":4309333648}"
DBG 2024-06-06 12:21:55.214-03:00 NewBE_callPrivateRPC                       topics="rpc" tid=1592317 file=core.nim:27 rpc_method=wakuext_checkPermissionsToJoinCommunity
DBG 2024-06-06 12:21:55.214-03:00 NewBE_callPrivateRPC                       topics="rpc" tid=1592316 file=core.nim:27 rpc_method=wakuext_checkAllCommunityChannelsPermissions
DBG 2024-06-06 12:21:56.018-03:00 NewBE_callPrivateRPC                       topics="rpc" tid=1592300 file=core.nim:27 rpc_method=accounts_getCollectiblePreferences
DBG 2024-06-06 12:21:56.091-03:00 NewBE_callPrivateRPC                       topics="rpc" tid=1592300 file=core.nim:27 rpc_method=accounts_getCollectiblePreferences
DBG 2024-06-06 12:21:56.689-03:00 NewBE_callPrivateRPC                       topics="rpc" tid=1592300 file=core.nim:27 rpc_method=wakuext_speedupArchivesImport
DBG 2024-06-06 12:21:56.787-03:00 NewBE_callPrivateRPC                       topics="rpc" tid=1592300 file=core.nim:27 rpc_method=accounts_getCollectiblePreferences
DBG 2024-06-06 12:21:57.591-03:00 NewBE_callPrivateRPC                       topics="rpc" tid=1592300 file=core.nim:27 rpc_method=accounts_getCollectiblePreferences
DBG 2024-06-06 12:22:15.380-03:00 NewBE_callPrivateRPC                       topics="rpc" tid=1592300 file=core.nim:27 rpc_method=wakuext_slowdownArchivesImport
DBG 2024-06-06 12:22:22.375-03:00 NewBE_callPrivateRPC                       topics="rpc" tid=1592300 file=core.nim:27 rpc_method=wakuext_getContactByID
DBG 2024-06-06 12:22:22.376-03:00 id doesn't have expected length            topics="contacts-service" tid=1592300 file=service.nim:417
DBG 2024-06-06 12:22:22.376-03:00 [threadpool task thread] initiating task   topics="task-threadpool" tid=1592325 file=threadpool.nim:54 messageType=AsyncGetRevealedAccountsArg:ObjectType threadid=1592325 task="{\"$type\":\"AsyncGetRevealedAccountsArg:ObjectType\",\"communityId\":\"0x021fbe1eebae9552401e7afb307c8a9cf78d1131ab4a76ef4635838e907f034eda\",\"memberPubkey\":\"0x040ad366b04230dea52b886d891047a184c196f08f93ee74dfd203a2e4b9978721a2fcd8cf91dbad239f6dc0d2f2bcb957efae0b8bad92fe1244eecbc62d92643d\",\"vptr\":105553181865472,\"slot\":\"onAsyncGetRevealedAccountsForMemberCompleted\",\"tptr\":4309894576}"
DBG 2024-06-06 12:22:22.376-03:00 [threadpool task thread] initiating task   topics="task-threadpool" tid=1592322 file=threadpool.nim:54 messageType=AsyncCheckPermissionsToJoinTaskArg:ObjectType threadid=1592322 task="{\"$type\":\"AsyncCheckPermissionsToJoinTaskArg:ObjectType\",\"communityId\":\"0x021fbe1eebae9552401e7afb307c8a9cf78d1131ab4a76ef4635838e907f034eda\",\"addresses\":[],\"vptr\":105553181865472,\"slot\":\"onAsyncCheckPermissionsToJoinDone\",\"tptr\":4309784232}"
DBG 2024-06-06 12:22:22.376-03:00 NewBE_callPrivateRPC                       topics="rpc" tid=1592325 file=core.nim:27 rpc_method=wakuext_getRevealedAccounts
DBG 2024-06-06 12:22:22.376-03:00 [threadpool task thread] initiating task   topics="task-threadpool" tid=1592321 file=threadpool.nim:54 messageType=AsyncCheckAllChannelsPermissionsTaskArg:ObjectType threadid=1592321 task="{\"$type\":\"AsyncCheckAllChannelsPermissionsTaskArg:ObjectType\",\"communityId\":\"0x021fbe1eebae9552401e7afb307c8a9cf78d1131ab4a76ef4635838e907f034eda\",\"addresses\":[],\"vptr\":105553181865216,\"slot\":\"onAsyncCheckAllChannelsPermissionsDone\",\"tptr\":4309333648}"
DBG 2024-06-06 12:22:22.376-03:00 NewBE_callPrivateRPC                       topics="rpc" tid=1592322 file=core.nim:27 rpc_method=wakuext_checkPermissionsToJoinCommunity
DBG 2024-06-06 12:22:22.376-03:00 NewBE_callPrivateRPC                       topics="rpc" tid=1592321 file=core.nim:27 rpc_method=wakuext_checkAllCommunityChannelsPermissions
DBG 2024-06-06 12:22:22.896-03:00 NewBE_callPrivateRPC                       topics="rpc" tid=1592300 file=core.nim:27 rpc_method=wakuext_speedupArchivesImport
DBG 2024-06-06 12:22:23.184-03:00 NewBE_callPrivateRPC                       topics="rpc" tid=1592300 file=core.nim:27 rpc_method=accounts_getCollectiblePreferences
DBG 2024-06-06 12:22:23.186-03:00 NewBE_callPrivateRPC                       topics="rpc" tid=1592300 file=core.nim:27 rpc_method=accounts_getCollectiblePreferences
DBG 2024-06-06 12:22:23.949-03:00 NewBE_callPrivateRPC                       topics="rpc" tid=1592300 file=core.nim:27 rpc_method=accounts_getCollectiblePreferences
DBG 2024-06-06 12:22:24.720-03:00 NewBE_callPrivateRPC                       topics="rpc" tid=1592300 file=core.nim:27 rpc_method=accounts_getCollectiblePreferences
```
geth.log
[geth.log.zip](https://github.com/user-attachments/files/15644816/geth.log.zip)

**Comment by osmaczko** [comment]
Thanks @Valentina1133 and @lukaszso for testing.

AD1.
This might be a token master client-side issue. Could you please check the member list on the control node side? If the member is on the list of the permissioned channel, then it means the bug is not related to reevaluation (or to my changes).

AD2.
This might be an issue with a missing encryption key. See details here: https://github.com/status-im/status-desktop/issues/14683#issuecomment-2133910622 under 'Current Method: Member list presence.' Another possibility is that after the airdrop, it can take up to 5 minutes for the reevaluation to start. Maybe you were not yet evaluated to be a token master? Could you please check the member list on the control node side? If the member is on the list, it means the bug is not related to reevaluation.

---

I'll briefly describe reevaluation so that you have a better overview of what might be related to this PR and what might not.

Reevaluation is a process that happens on the control node side. The control node iterates over permissions and members, checks on-chain if criteria are satisfied or not, and based on that, constructs the members list for each channel. Then this information is propagated to all members so that they can update their side.

The problem here is that members might have issues with properly interpreting the channel's state due to missing encryption keys, as mentioned in AD2. Until you receive that key, if the channel is private, you will not see this channel. As a consequence, you may think you don't have proper privileges. We are [working](https://github.com/status-im/status-desktop/issues/14931) on fixing that.

Moreover, reevaluation happens a maximum of once every 5 minutes, which means some actions (like airdrops or permission changes) could take effect with a delay.

In short, the source of truth is the control node. To check if reevaluation is working correctly, in most cases, it is enough to wait ~5 minutes after changes and check the members list of channels on its side.

**Comment by anastasiyaig** [comment]
thank you very much @osmaczko for detailed explanation! 
i think we can merge this PR now and @lukaszso and @virginiabalducci will log issues after rechecking the same scenarios in master

**Comment by osmaczko** [comment]
Thanks, @anastasiyaig. Would it be possible to re-test issue 1, please? Otherwise, if it really can't reevaluate members correctly, then the regression is too big to be fixed on master, I think. Better safe than sorry :sweat_smile:

**Comment by virginiabalducci** [comment]
> Thanks @virginiabalducci and @lukaszso for testing.
> 
> AD1. This might be a token master client-side issue. Could you please check the member list on the control node side? If the member is on the list of the permissioned channel, then it means the bug is not related to reevaluation (or to my changes).
> 

I checked the member list logged into the Community Owner's account and I don't see the user holding the token on the member list:

- Tabby Cat user is the owner
- VirDevTest is my user from the local build which had an updated pull request from yesterday, this user holds a token master and has admin access.

Screenshot 2024-06-07 at 9 36 25 AM

Test010 user holds the corresponding token and cannot see the channel.
Screenshot 2024-06-07 at 9 35 57 AM
Screenshot 2024-06-07 at 9 40 18 AM

> AD2. This might be an issue with a missing encryption key. See details here: [#14683 (comment)](https://github.com/status-im/status-desktop/issues/14683#issuecomment-2133910622) under 'Current Method: Member list presence.' Another possibility is that after the airdrop, it can take up to 5 minutes for the reevaluation to start. Maybe you were not yet evaluated to be a token master? Could you please check the member list on the control node side? If the member is on the list, it means the bug is not related to reevaluation.
> 
In this case, user Test004 holds TokenMaster and does not see any restricted channel nor has any admin permissions.

Screenshot 2024-06-07 at 9 42 17 AM

Screenshot 2024-06-07 at 9 42 30 AM

Owner (logged from control node)
Screenshot 2024-06-07 at 9 42 46 AM
Screenshot 2024-06-07 at 9 42 57 AM

I have been testing these accounts since yesterday (almost 24 hours), closing and re-opening app many times, and still I see these issues. 

However it is worth mentioning, that this only happens with Test004 account.

- Testing the same scenario with @lukaszso (his user receiving the token master on his end) he was able to see the changes almost immediately, with user being upgraded to Admin and able to create channels, see all channels, mint and airdrop tokens, etc. His admin user is not showing currently on the restricted channels because we also tested "owner remotely destructing tokenmaster" and he lost admin permissions almost immediately.
- I tested this on master (pull request from yesterday) and I was able to get the tokenmaster, see the hidden channels, etc as expected from an Admin user
- **I also created a new test account Test013, sent the tokenmaster. In less than 5 minutes this user gained admin permissions, can access all channels. **This leads me to believe that there may be a specific issue with Test004 account.** I will continue to test this account and report any findings in a new issue if needed.**

Screenshot 2024-06-07 at 10 02 22 AM
Screenshot 2024-06-07 at 10 02 42 AM
Screenshot 2024-06-07 at 10 03 07 AM

As per Issue 1, if I don't see the users holding the token in the channel list, is it under the scope of this ticket? Thanks!

**Comment by osmaczko** [comment]
> As per Issue 1, if I don't see the users holding the token in the channel list, is it under the scope of this ticket? Thanks!

Yes, definitely, I am looking into it right now.

**Comment by osmaczko** [comment]
> > As per Issue 1, if I don't see the users holding the token in the channel list, is it under the scope of this ticket? Thanks!
> 
> Yes, definitely, I am looking into it right now.

@virginiabalducci I say the famous "it works on my side" :sweat_smile: 


https://github.com/status-im/status-desktop/assets/33099791/6b4ed194-51ac-4cf5-81ef-b7a210471e9d

**Comment by osmaczko** [comment]
After an extensive testing session with @virginiabalducci, we concluded that the reevaluation is working fine. However, there are other side issues causing confusion. The most perplexing issue is that one of the existing users (user010) is never evaluated, while other members are. It appears there is a problem with the addresses shared by this user, as confirmed by the following screenshot:
image
The addresses cannot be recognized as belonging to user010. A separate issue will be created to investigate this further.
